### PR TITLE
[master] HELP-30998: fix timezone in teletype fax template, unified get timezone

### PR DESCRIPTION
**Pull request by icehess**
* make sure the code paths from `kzd_fax_box`, `kzd_device`, etc... to get timezone resulted in non undefined timezone
* fix `teletype_util:fix_timestamp/3`
* pretty print caller, caller id numbers
* fix mod_config_cat in templates

### CODE REVIEW

File: applications/callflow/src/cf_util.erl [hunk 1]
```diff
@@ -796,18 +796,11 @@ process_event(Call, NoopId, JObj) ->
 -spec get_timezone(kz_json:object(), kapps_call:call()) -> ne_binary().
 get_timezone(JObj, Call) ->
     case kz_json:get_ne_binary_value(<<"timezone">>, JObj) of
-        'undefined'   -> account_timezone(Call);
-        <<"inherit">> -> account_timezone(Call);  %% UI-1808
+        'undefined'   -> kz_account:timezone(kapps_call:account_id(Call));
+        <<"inherit">> -> kz_account:timezone(kapps_call:account_id(Call));
```
Comment (fenollp) [inline applications/callflow/src/cf_util.erl]: Please keep the comment, may be useful in the future :)

File: applications/callflow/src/cf_util.erl [hunk 2]
```diff
@@ -796,18 +796,11 @@ process_event(Call, NoopId, JObj) ->
...
         TZ -> TZ
     end.
 
--spec account_timezone(kapps_call:call()) -> ne_binary().
-account_timezone(Call) ->
-    case kz_account:fetch(kapps_call:account_id(Call)) of
-        {'ok', AccountJObj} -> kz_account:timezone(AccountJObj);
-        {'error', _E} -> kz_account:default_timezone()
```
Comment (fenollp) [inline applications/callflow/src/cf_util.erl]: Semantics are not preserved: this will now crash on error: https://github.com/2600hz/kazoo/blob/master/core/kazoo_documents/src/kz_account.erl?utf8=%E2%9C%93#L248

Was this change on purpose?

File: applications/crossbar/src/modules/cb_faxes.erl [hunk 3]
```diff
@@ -516,29 +516,36 @@ verify_number(Context) ->
...
-                                          ,{<<"fax_timezone">>, Timezone}
-                                          ]
+                                          ] ++ maybe_add_timezone(Context)
                                          ,cb_context:doc(Context)
                                          )
                       ).
 
+-spec maybe_add_timezone(cb_context:context()) -> kz_proplist().
+maybe_add_timezone(Context) ->
+    maybe_add_timezone(Context, kz_json:get_value(<<"timezone">>, cb_context:doc(Context))).
```
Comment (fenollp) [inline applications/crossbar/src/modules/cb_faxes.erl]: "maybe_" looks like it can be dropped.

File: core/kazoo_documents/src/kz_account.erl [hunk 4]
```diff
@@ -241,18 +241,35 @@ set_language(JObj, Language) ->
...
 %%--------------------------------------------------------------------
--spec timezone(ne_binary() | doc()) -> api_ne_binary().
--spec timezone(ne_binary() | doc(), Default) -> ne_binary() | Default.
-timezone(AccountId)
-  when is_binary(AccountId) ->
-    {'ok', JObj} = fetch(AccountId),
-    timezone(JObj);
+-spec timezone(api_ne_binary() | doc()) -> ne_binary().
+timezone('undefined') ->
+    ?DEFAULT_TIMEZONE;
```
Comment (fenollp) [inline core/kazoo_documents/src/kz_account.erl]: Oh right then disregard semantics comment! :)

File: core/kazoo_documents/src/kz_device.erl [hunk 5]
```diff
@@ -303,42 +304,12 @@ timezone(Box, Default) ->
...
-                                  )
-    of
-        {'ok', OwnerJObj} -> OwnerJObj;
-        {'error', 'not_found'} -> 'undefined'
-    end.
-
-owner_timezone(Box, Default, OwnerJObj) ->
-    case kzd_user:timezone(OwnerJObj, 'undefined') of
-        'undefined'   -> account_timezone(Box, Default);
-        <<"inherit">> -> account_timezone(Box, Default);  %% UI-1808
```
Comment (fenollp) [inline core/kazoo_documents/src/kz_device.erl]: This scares me. Remember this @k-anderson ?

Comment (icehess) [inline core/kazoo_documents/src/kz_device.erl]: why?

Comment (icehess) [inline core/kazoo_documents/src/kz_device.erl]: Both `kzd_user` and `kz_account` check `inherit` anyway.
